### PR TITLE
Enable CI builds with postgresql and mariadb

### PR DESCRIPTION
**Pull request by zorun**
This is still a WIP to see if it works.

### CODE REVIEW

**Comment by indatwood** [comment]
Hey @zorun what should we do with this? Was it working? Should we merge it, or close it? Thanks for the heads up!

**Comment by zorun** [comment]
> Hey @zorun what should we do with this? Was it working? Should we merge it, or close it? Thanks for the heads up!

Hey, sorry for leaving this in a WIP state.  I just rebased it (it needed some manual fix-up).

As far as I remember, it wasn't merged because the tests were failing with mysql (see #632). We need to fix that issue before merging, otherwise all PRs would appear to be faulty.

**Comment by zorun** [comment]
This should fail (because I explictly trigger a known bug): https://travis-ci.org/github/spiral-project/ihatemoney/builds/757617787

This is supposed to work but will probably fail because of #632 : https://travis-ci.org/github/spiral-project/ihatemoney/builds/757618257

**Comment by almet** [comment]
Hey, if the tests are failing with this, and it's the expected behavior, then I think we should merge it and fix the bug that this points out. Right?

So, merge time?

**Comment by zorun** [comment]
> Hey, if the tests are failing with this, and it's the expected behavior, then I think we should merge it and fix the bug that this points out. Right?
> 
> So, merge time?

Well, it depends if fixing the MySQL bug takes 1 day or 1 year :)

If we merge this while not fixing the bug, all pull requests will have failing CI. It's not very nice for people proposing patches.

**Comment by Glandos** [comment]
Is it possible to replace `service: mysql` with `mariadb` to test it as noted in https://github.com/spiral-project/ihatemoney/issues/632#issuecomment-817342784 ?

**Comment by zorun** [comment]
I've added support for the github CI, and switched to mariadb instead of mysql.

I'm not 100% satisfied, because the postgresql and mysql containers are started in all jobs (even those using sqlite). But it only means that the CI is a bit longer than necessary, and fixing it would involve writing multiple actions with some kind of inheritance, and I'm too lazy to learn how to do that.

So, it's good to go for a review @Glandos !

File: .travis.yml [hunk 1]
```diff
@@ -5,8 +5,18 @@ python:
...
+  - psql -c 'create database travis_ci_test;' -U postgres
+  - mysql -e 'CREATE DATABASE travis_ci_test;'
 script: tox
 before_install:
   - python -m pip install --upgrade pip virtualenv
 install:
   - pip install tox-travis
+services:
+  - mysql
+  - postgresql
```
Comment (Glandos) [inline .travis.yml:22]: Is this a leftover of an old commit?
I don't have any objection against maintaining old Travis config, but it references mysql, so I guess this should be either dropped or updated.

**Review by Glandos — CHANGES_REQUESTED** [review]
See comment on .travis.yml

File: .travis.yml (hunk 1 above)
Comment (zorun) [reply .travis.yml:22]: Right, this is a commit I did when we were using Travis.

I have now removed all Travis changes.

**Comment by zorun** [comment]
Uh, weird, I just rebased and there's a new postgresql error:

```
psycopg2.errors.GroupingError: column "bill.id" must appear in the GROUP BY clause or be used in an aggregate function
```

**Comment by zorun** [comment]
Ah, it's the new check from #661 . I should have merged before you @Glandos ^^

```
return self.get_bills_unordered().group_by(Bill.original_currency).count() > 1
```

I'm looking at it.

**Pull request merged**